packages/vaadin-element-mixin/vaadin-dir-mixin.js

```diff
@@ -22,6 +22,8 @@ const directionObserver = new MutationObserver(directionUpdater);
 directionObserver.observe(document.documentElement, { attributes: true, attributeFilter: ['dir'] });
 
 const alignDirs = function (element, documentDir) {
+  const elementDir = element.getAttribute('dir');
```

Comment:
One more idea on how to avoid `getAttribute` here.
It looks like `alignDirs` is called in two places:

1. In `connectedCallback` if no attribute is set.
2. In `attributeChangedCallback` where we know new value.

So I would expect us to add a third parameter to `alignDirs` so it would look like this:

```js
const alignDirs = function (element, documentDir, elementDir = '') {
```

Then we can pass `elementDir` from the `attributeChangedCallback` and use it here.

Reply:
Actually, there is one more place in the MutationObserver callback that could still need `getAttribute`.

Reply:
@web-padawan I don't feel too strongly about this, but I would prefer to keep as is:
- the function is better encapsulated and doesn't rely on callers passing the correct value
- ATM in 2 out of 3 usages we have to call `getAttribute` anyway

What would be the motivation for this change, do you see a performance improvement here?

Reply:
I'm not sure if there is going to be a significant impact on the performance.

- When a single node changes, it probably does not harm to have another `getAttribute` call.
- And when a `dir` changes on the document level, there will be many `getAttribute` calls anyways.

I have put together a branch with some suggestions, feel free to decide if you want to include them or now.
Please see the diff here: https://github.com/vaadin/web-components/compare/fix/dir-mixin-native-dir-api...dir-reduce-dom-calls

Reply:
OK, with the default param it's a lot better. Thanks, I merged it.